scheduler/reconcile_test.go

```diff
@@ -5632,18 +5705,25 @@ func TestReconciler_Disconnected_Client(t *testing.T) {
 						replacement.Metrics.ScoreMetaData[0].NormScore = replacement.Metrics.ScoreMetaData[0].NormScore + tc.nodeScoreIncrement
 					}
 
-					replacements = append(replacements, replacement)
-
 					// If we want to test intermediate replacement failures simulate that.
 					if tc.failReplacement {
 						replacement.ClientStatus = structs.AllocClientStatusFailed
-						nextReplacement := replacement.Copy()
-						nextReplacement.ID = uuid.Generate()
-						nextReplacement.ClientStatus = structs.AllocClientStatusRunning
-						nextReplacement.PreviousAllocation = replacement.ID
-						replacement.NextAllocation = nextReplacement.ID
-						replacements = append(replacements, nextReplacement)
+
+						if tc.replaceFailedReplacement {
+							nextReplacement := replacement.Copy()
+							nextReplacement.ID = uuid.Generate()
+							nextReplacement.ClientStatus = structs.AllocClientStatusRunning
+							nextReplacement.DesiredStatus = structs.AllocClientStatusRunning
+							nextReplacement.PreviousAllocation = replacement.ID
+
+							replacement.NextAllocation = nextReplacement.ID
+							replacement.DesiredStatus = structs.AllocDesiredStatusStop
```

Comment:
This was missing but I'm pretty sure we stop the previous allocation once it has been replaced 🤔